feat: add advanced logging

stefina opened the pull request.

stefina committed:
feat: add advanced logging

stefina committed:
feat: make propagate-option configurable

stefina committed:
fix: use log_level as int

stefina committed:
fix: add dummy test

fix: add dummy test

bellisk commented on README.md:
```diff
@@ -55,6 +55,14 @@ Make sure to configure the following configuration-entries to set up logging wit
 	ckanext.sentry.log_level = WARNING
 
+    # Configure loggers, separated with spaces, for sentry:
+	# (optional, default: ["", "ckan", "ckanext", "sentry.errors"]).
```
Would be good to have this in the same format as the line below.

bellisk commented on README.md:
```diff
@@ -55,6 +55,14 @@ Make sure to configure the following configuration-entries to set up logging wit
+	# (optional, default: ["", "ckan", "ckanext", "sentry.errors"]).
+	ckanext.sentry.loggers = ckan ckanext ckanext.dcat ckanext.scheming.plugin
+
+    # Enable propagated logging by setting to `True` or avoid duplicate events from parent loggers:
```
This line isn't clear to me, sorry. :sweat_smile:

stefina committed:
docs: improve config-variables documentation

stefina committed:
fix: don’t override the level of the logger

We don’t want to interfere with the initially set log-level and leave it as is. The log level will only apply to the Sentry-Handlers.

stefina committed:
feat: add more detailed documentation about configuration possibilities

stefina committed:
refactor: rename configuration to better match its function

This configuration entry enables the entire plugin, so you could have the plugin activated but only when setting it to `True` it will actually initialize the middleware.

stefina committed:
feat: make it possible to configure global error logging separately

Global error logging will log all messages of a defined level to sentry.
If you want to only use sentry for debugging a specific plugin, it is useful to deactivate global error logging.

stefina committed:
fix: improve documentation of conf-entry
